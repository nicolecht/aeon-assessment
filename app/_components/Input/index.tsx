export default function Input({
  className,
  ...props
}: React.InputHTMLAttributes<HTMLInputElement>) {
  return (
    <input
      className={`p-2 border rounded-lg w-full outline-none border-neutral-300 focus:border-neutral-500 ${className}`}
      {...props}
    />
  );
}
