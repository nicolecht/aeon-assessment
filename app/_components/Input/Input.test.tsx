import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Input from ".";

describe("Input component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Input placeholder="Enter text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with provided className", () => {
    render(<Input className="custom-class" />);
    const inputElement = screen.getByRole("textbox");
    expect(inputElement).toHaveClass("custom-class");
  });

  it("accepts user input", async () => {
    render(<Input />);
    const inputElement = screen.getByRole("textbox");

    await userEvent.type(inputElement, "Hello World");

    expect(inputElement).toHaveValue("Hello World");
  });

  it("calls onChange when typing", async () => {
    const handleChange = jest.fn();
    render(<Input onChange={handleChange} />);
    const inputElement = screen.getByRole("textbox");

    await userEvent.type(inputElement, "test");

    expect(handleChange).toHaveBeenCalledTimes(4); // "t", "e", "s", "t"
  });
});
