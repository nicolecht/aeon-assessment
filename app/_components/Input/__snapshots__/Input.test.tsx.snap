// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="p-2 border rounded-lg w-full outline-none border-neutral-300 focus:border-neutral-500 undefined"
    placeholder="Enter text"
  />
</DocumentFragment>
`;
